addons/account/models/account_move.py

```diff
@@ -1410,7 +1410,8 @@ def _compute_quick_encoding_vals(self):
     def _compute_duplicated_ref_ids(self):
         move_to_duplicate_move = self._fetch_duplicate_supplier_reference()
         for move in self:
-            move.duplicated_ref_ids = move_to_duplicate_move.get(move, self.env['account.move'])
+            # Uses move._origin.id to handle records in edition/existing records and 0 for new records, for duplicated reference ID assignment.
+            move.duplicated_ref_ids = move_to_duplicate_move.get(self.env["account.move"].browse(move._origin.id or 0), self.env['account.move'])
```

Comment:
Explicitly browsing 0 is ugly.
But anyway, isn't _origin a recordset already ? Why not doing this, then ?

```suggestion
            move.duplicated_ref_ids = move_to_duplicate_move.get(move._origin, self.env['account.move'])
```

I think you could even write [] instead of self.env['account.move'] ; to check.

Reply:
Alright, I won't fight on that, let's agree to disagree ^^ No need to change :)